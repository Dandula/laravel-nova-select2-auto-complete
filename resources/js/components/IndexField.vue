<template>
    <div v-if="field.value">
        <div v-for="option in field.options">
            <template v-if="field.showAsLink">
                <router-link
                        v-if="valueExist(field.value, option.id)"
                        :to="{
                            name: 'detail',
                            params: {
                                resourceName: field.linkToResource || resourceName,
                                resourceId: option.id
                            }
                        }"
                        class="no-underline dim text-primary font-bold"
                >
                    {{ labelFor(option.text) }}
                </router-link>
            </template>
            <template v-else>
                <span v-if="valueExist(field.value, option.id)" class="text-left" v-html="labelFor(option)"></span>
            </template>
        </div>
    </div>

    <span v-else class="whitespace-no-wrap text-left">&mdash;</span>
</template>

<script>
    export default {
        props: ['resourceName', 'field'],
        methods: {
            labelFor(text) {
                return text.replace(/(?:\r\n|\r|\n)/g, ' ')
            },
            valueExist(fieldValue, optionValue) {
                fieldValue = Array.isArray(fieldValue) ? fieldValue : [fieldValue]

                return fieldValue.indexOf(optionValue) !== -1
            }
        }
    }
</script>
