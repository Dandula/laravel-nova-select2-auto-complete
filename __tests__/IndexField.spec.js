import {mount, shallowMount, createLocalVue} from '@vue/test-utils'
import PortalVue from 'portal-vue'
import IndexField from '../resources/js/components/IndexField'

const localVue = createLocalVue()
localVue.use(PortalVue)

describe('IndexField', () => {
    test('it renders correctly without field.value', () => {
        const wrapper = mount(IndexField, {
            localVue,
            propsData: {
                resourceName: 'category',
                field: {
                    value: null
                }
            },
        })

        expect(wrapper).toMatchSnapshot()
    })

    test('it renders correctly with field.value + options', () => {
        const wrapper = mount(IndexField, {
            localVue,
            propsData: {
                resourceName: 'category',
                field: {
                    value: 3,
                    options: [
                        {id: 1, text: 'One'},
                        {id: 2, text: 'Two'},
                        {id: 3, text: 'Three'}
                    ]
                }
            }
        })

        expect(wrapper).toMatchSnapshot()
    })

    test('it renders correctly with field.value + options + showAsLink', () => {
        const wrapper = mount(IndexField, {
            localVue,
            propsData: {
                resourceName: 'category',
                field: {
                    value: 3,
                    options: [
                        {id: 1, text: 'One'},
                        {id: 2, text: 'Two'},
                        {id: 3, text: 'Three'}
                    ],
                    showAsLink: true
                }
            },
            stubs: ['router-link', 'router-view']
        })

        expect(wrapper).toMatchSnapshot()
    })
})
