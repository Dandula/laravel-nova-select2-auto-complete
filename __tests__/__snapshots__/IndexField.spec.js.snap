// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexField it renders correctly with field.value + options + showAsLink 1`] = `
<div>
  <div>
    <!---->
  </div>
  <div>
    <!---->
  </div>
  <div>
    <router-link-stub to="[object Object]" class="no-underline dim text-primary font-bold">
      Three
    </router-link-stub>
  </div>
</div>
`;

exports[`IndexField it renders correctly with field.value + options 1`] = `
<div>
  <div>
    <!---->
  </div>
  <div>
    <!---->
  </div>
  <div><span class="text-left">Three</span></div>
</div>
`;

exports[`IndexField it renders correctly without field.value 1`] = `<span class="whitespace-no-wrap text-left">â€”</span>`;
